### PR TITLE
newdevice: Added detection for Hirschmann Railswitch #6161

### PR DESCRIPTION
**Pull request by mail4git38**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/discovery/os/hirschmann.inc.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
```
Comment (murrant) [inline includes/discovery/os/hirschmann.inc.php]: Remove this file, it is handled by the yaml.

File: includes/definitions/hirschmann.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+os: hirschmann
+text: 'Hirschmann Railswitch'
+type: network
+icon: hirschmann
+over:
+    - { graph: device_bits, text: 'Device Traffic' }
+    - { graph: device_processor, text: 'CPU Usage' }
+    - { graph: device_mempool, text: 'Memory Usage' }
+discovery:
+    - sysDescr:
```
Comment (murrant) [inline includes/definitions/hirschmann.yaml]: Please use sysObjectId instead of sysDescr.

**Comment by mail4git38** [comment]
Sorry, I'm a new user with git :-(

The requested changes are done on my dev server but I can't push my commits to issue-6161.

Here is the message I have : 'updates were rejected because the tip of your current branch is behind its remote counterpart'

How can I push my changes ?

File: includes/polling/os/hirschmann.inc.php [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+$hardware = trim(snmp_get($device, "hmPNIOOrderID.0", "-OQv", "HMPRIV-MGMT-SNMP-MIB"), '"');
```
Comment (mail4git38) [inline includes/polling/os/hirschmann.inc.php]: trim() is indeed not needed here and in the lines below. I remove it.

File: includes/discovery/os/hirschmann.inc.php (hunk 1 above)
Comment (mail4git38) [inline includes/discovery/os/hirschmann.inc.php]: This file is an old file from a previous LibreNMS installation. I've removed it.

File: includes/definitions/hirschmann.yaml (hunk 2 above)
Comment (mail4git38) [inline includes/definitions/hirschmann.yaml]: sysDescr is now replaced by sysObjectId.

**Comment by murrant** [comment]
@mail4git38 sorry for the late response.  Run `git pull` first to pull down remote changes.

Sorry about that, I was fixing merge conflict.

**Comment by mail4git38** [comment]
Hello @murrant. Thanks for your response and no problem with the delay.

I've tried to run `git pull upstream master` command before `git push origin issue-6161` but I still have the same issue (rejected updates).

**Comment by laf** [comment]
You need to run `git pull` or if that doesn't work `git pull origin issue-6161` (i've assumed your remote is called origin - replace that name if it's not)

**Comment by mail4git38** [comment]
Thanks a lot for your help on git for my first contribution. 

I successfully pushed my new commits to the issue 6161.

**Comment by laf** [comment]
Only two things from me (aside from the merge conflict).

You have two files for state, one in `includes/discovery/sensors/states/` which is the old location and needs to be deleted.

The other is you've got the same `$index` value in all of the state support so this shouldn't be working. You would be better off updating these to use `name.$index` ie.: `hmPowerSupplyStatus.0`, `hmPowerSupplyStatus.1`, etc.

**Comment by laf** [comment]
Just the issue with state indexes now and then good for a merge.

**Comment by laf** [comment]
@mail4git38 Thanks for submitting this. Unfortunately you've removed yourself from the AUTHORS.md at the end of this PR. Please submit a new pull request adding yourself back in and agreeing to the contributors agreement otherwise we will have to revert this pull request.

**Pull request merged**